### PR TITLE
test: add tests for pkg/engine/engine.go

### PR DESCRIPTION
**Pull request by tariq1890**
This is a more resilient and consistent version of PR #787. 

This has test cases that are resilient to non-deterministic order of map key-value pairs

### CODE REVIEW

File: pkg/engine/engine_test.go [hunk 1]
```diff
@@ -586,3 +620,158 @@ func TestGetDataDisks(t *testing.T) {
...
+		CustomNodeLabels: map[string]string{
+			"foo":   "bar",
+			"abc":   "xyz",
+			"lorem": "ipsum",
+		},
+	}
+
+	actual := getDCOSWindowsAgentCustomAttributes(profile)
+
+	if !strings.Contains(actual, "os:Windows;public_ip:yes;") {
```
Comment (tariq1890) [inline pkg/engine/engine_test.go]: Lines 734-742 are the improvements

**Comment by jackfrancis** [comment]
lgtm pending unit tests

**Comment by tariq1890** [comment]
68.99% . Where did that 0.01 go :'( ?

**Pull request merged**